Makefile

```diff
@@ -57,17 +57,19 @@ check: gopath
 
 .PHONY: lint
 lint: gopath
-	@gometalinter.v1 --deadline=10m --tests --vendor --disable-all \
+	@extra_args= ;\
+	[ -z "${CYCLO_MAX}" ] || extra_args="--enable=gocyclo --cyclo-over=${CYCLO_MAX}" ;\
+	gometalinter.v1 --deadline=10m --tests --vendor --disable-all \
```

Comment:
This won't run if gometalinter is not installed to /usr/bin, should probably use `$(GOPATH)/bin/gometalinter.v1`

Reply:
It seems to work for the travis. I have $GOPATH/bin as part of my $PATH -- I guess by now most of us do.

In the case we are using a "new GOPATH" we'll need to keep track of the old one for the sake of this command line call. So, before I enter that rabbit hole, is this a real issue for you or an hypothetical case?

Reply:
Yes this works for Travis, but it doesn't work for a local build which doesn't install gometalinter or add GOPATH to $PATH. If $GOPATH/bin is already in your path just calling it with $GOPATH/bin/<program> shouldn't cause any problems and cover both cases IMO. It's already being used for the install target too  :)

Reply:
The compliant target is not needed anymore with this change since fmt and vet will be run, and it exits appropriately with an exit by default.

Reply:
You can not in general use $GOPATH/bin/metalinter.v1. This is because GOPATH, like PATH, is a colon separated list of directories. I agree that the common case is for it to be just a single one, but just as you can't say $PATH/ls, you can't append bin/metalinter.v1 to $GOPATH.

If you want to do something like that, this, then I suggest the following is the correct way to do it. Works with an empty GOPATH, one with a single directory in it, works with multiple directories in it. 

     _G=${GOPATH:-$HOME/go}:
     PATH=${_G//:/\/bin:}$PATH metalinter.v1

(works by setting _G to the GOPATH or the default one if it is not set, and appends a colon, then changes all colons to  slash bin colon, and invokes metalinter.v1 with this prepended onto the PATH so it is found first.

Reply:
> Yes this works for Travis, but it doesn't work for a local build which doesn't install gometalinter or add GOPATH to $PATH.

This is more about #51. It wasn't my intention to go all the way to automatically install gometalinter -- since it is a bit of a rabbit hole to deal with all the cases. Note that before the commit in question, we had to copy/paste to run these tests :-P

Having a target for installing the lint tools could be useful, but I thought it was less important than 'make lint' itself.

Reply:
> This is because GOPATH, like PATH, is a colon separated list of directories.

Yep. The Makefile still uses it in a way that is not 100% guaranteed to work (but was there before). We'll fix that eventually.

All this becomes more complicated because we are using this Makefile both as "helper for developer that follows Go conventions" and "helper for packaging that don't care about go conventions". Wondering if maybe the real fix is to teach autospec a bit more about go. E.g. something like Travis' go_import_path keyword would make things easier.